### DISCUSSION TITLE
How to import python class

### DISCUSSION THREAD

**Discussion by tqq1994516** [modular/modular#807] [Q&A] [answered]
I try to get the class in python lib, the code is as follows
```python
from python import Python

fn main():
    try:
        let pydantic = Python.import_module("pydantic")
        print(pydantic.BaseModel)
    except e:
        print("failed to show plot:", e.value)
```
but get this error
```bash
failed to show plot: An error occurred in Python.
```

**Comment by rarebreed** [comment]
Did you `pip install pydantic`?  I generally always create a virtualenv first, activate it, and run my mojo.  If you're in VS code, make sure you select the virtualenv also

**Reply by tqq1994516** [reply]
I can print `pydantic` variable object，but if me print `pydantic.BaseModel`, I will get this error.

**Reply by tqq1994516** [reply]
I can print `pydantic` variable object，but if me print `pydantic.BaseModel`, I will get this error.

**Comment by rarebreed** [comment, marked as answer]
> I can print `pydantic` variable object，but if me print `pydantic.BaseModel`, I will get this error.

In the except clause, I just did `print(e.value)` and I got this error (using python 3.11.5)

```
❯ mojo run main.mojo
No string method found on the python object.
```

so, mojo's `print` is not the same as python's `print`.  In python, the `print()` function will implicitly do something like this

```python
# this is python
import pydantic as pd
print(str(pd.BaseModel))
```

From the python docs

> All non-keyword arguments are converted to strings like [str()](https://docs.python.org/3/library/stdtypes.html#str) does and written to the stream

In mojo, the [print](https://docs.modular.com/mojo/stdlib/builtin/io.html#print) function works with mojo types, though it does have an overloaded type for `object`.  My guess is that the mojo `print` doesn't do the implicit call to `str`.  Not sure if that is a bug, or a "not implemented yet" feature.

But you can try this workaround

```
fn main():
    try:
        let pydantic = Python.import_module("pydantic")
        let bi = Python.import_module("builtins")
        print(bi.str(pydantic.BaseModel))
        # or bi.print(pd.BaseModel)
    except e:
        print("failed to show plot:", e.value)

**Reply by tqq1994516** [reply]
What you said is indeed correct, and the `pydantic.BaseModel` cannot be instantiated directly, which makes me think that the class cannot be imported. I changed the class instantiation and use of another library and there is no problem.
